### ISSUE TITLE
fish error when ``programs.fish.vendor.config.enable = true``

### ISSUE THREAD

**Issue by magnetophon** [NixOS/nixpkgs#261777]
### Describe the bug
When ``programs.fish.vendor.config.enable = true``, I get the following error at every fish prompt:
```
fish: Unknown command: fzf_key_bindings
/run/current-system/sw/share/fish/vendor_conf.d/load-fzf-key-bindings.fish (line 1):
fzf_key_bindings
^~~~~~~~~~~~~~~^
from sourcing file /run/current-system/sw/share/fish/vendor_conf.d/load-fzf-key-bindings.fish
        called on line 248 of file /nix/store/mnmy8xws1zxblajvnal6v11i32ji4qp6-fish-3.6.1/share/fish/config.fish
```
When I set it to false, the error goes away.

### Steps To Reproduce
1. set ``programs.fish.vendor.config.enable = true`` in your config. (this is the default.
2. open the fish shell in a terminal
3. see the error

### Expected behavior
No error.

### Notify maintainers
@cole-h @winter @srapenne

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.7, NixOS, 23.11 (Tapir), 23.11pre536534.ca012a02bf83`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(root): `"nixos, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by goofansu** [comment]
I encountered the same problem after updating packages with:
```shell
cd nix-config
nix flake update
darwin-rebuild switch --flake .
```

But starting new shells don't have the problem, so I restarted all existing shells.

nix-info:
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.6.0, macOS 12.7`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.17.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by magnetophon** [comment]
@goofansu Thanks for the reply.
I tried to reproduce your findings, but ran into https://github.com/NixOS/nixpkgs/issues/262266, so I can't test.

**Comment by travis-anderson** [comment]
I think this was addressed here yesterday: https://github.com/NixOS/nixpkgs/commit/a20c614010580dfb6ac182bdb3661777bbe50955

**Comment by simonbcn** [comment]
I have upgraded the system from 23.05 to unstable and now I have the same problem as _magnetophon_.

```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.7, NixOS, 23.11 (Tapir), 23.11pre536534.ca012a02bf83`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(root): `"nixos"`
 - channels(juan): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Parts of `configuration.nix` where I refer to fish and its plugins:
```
....
  users = {
    users.juan.isNormalUser = true;
    users.juan.description = "Juan";
    users.juan.extraGroups = [ "networkmanager" "wheel" ];
    users.juan.packages = with pkgs; [
...
    ];

    defaultUserShell = pkgs.fish;
  };
...
  environment = {

    systemPackages = with pkgs; [
      vscodium
      fishPlugins.done
      fishPlugins.tide
      fishPlugins.fzf-fish
      fzf
      fishPlugins.grc
      grc
      fishPlugins.colored-man-pages
      nil
      nixpkgs-fmt
      git
      nodejs
      plocate
      fd
      wget
    ];

    shells = with pkgs; [ fish ];
  };

  programs.fish = {
    enable = true;
    interactiveShellInit = ''
      set fish_greeting # Disable greeting
    '';
  };
....
```

**Comment by goofansu** [comment]
@simonbcn Just wait for https://github.com/NixOS/nixpkgs/pull/261778 backported to unstable.

**Comment by SuperSamus** [comment]
Using `nixpkgs-unstable` (which has landed the PR), the issue has been fixed specifically for fzf, but not for tide.
While the custom prompt is working, there's the error:
```
fish: Unknown command: _tide_item_virtual_env
/run/current-system/sw/share/fish/vendor_functions.d/_tide_1_line_prompt.fish (line 11): 
        _tide_item_$item
        ^~~~~~~~~~~~~~~^
in function '_tide_1_line_prompt'
in command substitution
fish: Unknown command: _tide_item_chruby
/run/current-system/sw/share/fish/vendor_functions.d/_tide_1_line_prompt.fish (line 11): 
        _tide_item_$item
        ^~~~~~~~~~~~~~~^
in function '_tide_1_line_prompt'
```

**Comment by R-VdP** [comment]
> @simonbcn Just wait for #261778 backported to unstable, but I didn't see the backport PR.

A backport should not be needed, since 23.05 is still on fzf 0.40 which should not be affected. Only unstable was affected.

**Comment by pbek** [comment]
Thank you, @R-VdP! Have to play with that more, the config caused some errors for me...

**Comment by R-VdP** [comment]
@pbek oh sorry, my fault, the PR patch would need to be applied to nixpkgs as a whole, not the fzf source, you can do it like this though:

```nix
pkgs.fzf.overrideAttrs (final: prev: {
  postInstall = (prev.postInstall or "") + ''
    cat << EOF > $out/share/fish/vendor_conf.d/load-fzf-key-bindings.fish
      status is-interactive; or exit 0
      fzf_key_bindings
    EOF
  '';
});
```

**Comment by pbek** [comment]
Didn't work either 😅

**Comment by goofansu** [comment]
@R-VdP Thanks, it works.

@pbek  Do you use home-manager? If so, it works like:

```nix
{ pkgs, ... }:

{
  programs.fzf = {
    enable = true;
    package = pkgs.fzf.overrideAttrs (final: prev: {
      postInstall = (prev.postInstall or "") + ''
        cat << EOF > $out/share/fish/vendor_conf.d/load-fzf-key-bindings.fish
          status is-interactive; or exit 0
          fzf_key_bindings
        EOF
      '';
    });
  };
}
```

**Comment by pbek** [comment]
@goofansu, yes, I do use home-manager. 😁  Thank you, that worked!
@R-VdP, sorry for messing up the GitHub issue. 😅

**Comment by SuperSamus** [comment]
In reply to my previous message, I fixed it by deleting `~/.config/fish/fish_variables`.

**Comment by pbek** [comment]
It's fixed for me now in nixpkgs unstable! 👍🏻

**Comment by R-VdP** [comment]
Fixed in #261778
